### PR TITLE
added proper group name & field name in system.xml for creditmemo feature

### PR DESCRIPTION
**Pull request by viral-wagento**
### Description

Added proper group name and field name for system configurations regarding creditmemo feature for zero order total, for below files:

git add app/code/Magento/Sales/Model/Order/Creditmemo.php
git add app/code/Magento/Sales/Test/Unit/Model/Order/CreditmemoTest.php
git add app/code/Magento/Sales/etc/adminhtml/system.xml
git add app/code/Magento/Sales/etc/config.xml

Actually I have developed same feature in PR: 16005 and then submitted required typos in 19288.
As per discussed with @orlangur in slack channel - we should add field name for both the fields.

### Reason why we should change the field name:

group id was  = zerograndtotal_creditmemo  - actually group id should be general name
field id was = allow_zero_grandtotal - field id should be field specific.
In above case both names seems not appropriate.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @viral-wagento. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by viral-wagento** [comment]
@orlangur  I have made changes in system.xml for sales module as per our discussion.
Can you please review it once?
Thanks!

**Comment by torhoehn** [comment]
@viral-wagento Please solve merge conflicts.

**Comment by viral-wagento** [comment]
@torhoehn, I will work on this and update soon

File: app/code/Magento/Sales/etc/adminhtml/system.xml [hunk 1]
```diff
@@ -48,9 +48,9 @@
                     <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                 </field>
             </group>
-            <group id="zerograndtotal_creditmemo" translate="label" type="text" sortOrder="30" showInDefault="1" showInWebsite="1" showInStore="1">
+            <group id="allow_zero_grandtotal" translate="label" type="text" sortOrder="30" showInDefault="1" showInWebsite="1" showInStore="1">
```
Comment (sidolov) [inline app/code/Magento/Sales/etc/adminhtml/system.xml:51]: Unfortunately, we are not allowed to remove/rename configuration according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/). Any other 3d party extension that depends on `sales/zerograndtotal_creditmemo/allow_zero_grandtotal` config will be broken.

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by orlangur** [comment]
@viral-wagento unfortunately this could only be changed before your changes were released :(

**Pull request closed without merging**